perun-core/src/main/java/cz/metacentrum/perun/core/bl/UsersManagerBl.java

```diff
@@ -1213,6 +1216,58 @@ void deletePassword(PerunSession sess, User user, String loginNamespace)
 	 */
 	void checkReservedLogins(PerunSession sess, String namespace, String login, boolean ignoreCase) throws AlreadyReservedLoginException;
 
+	/**
+	 * Returns all pairs of blocked login in namespace (if namespace is null, then this login is blocked globally)
+	 *
+	 * @param sess
+	 */
+	List<Pair<String, String>> getAllBlockedLoginsInNamespaces(PerunSession sess);
+
+	/**
+	 * Return true if login is blocked (globally - for all namespaces per instance OR for some namespace), false if not
+	 *
+	 * @param sess
+	 * @param login login to check
+	 */
+	boolean isLoginBlocked(PerunSession sess, String login);
+
+	/**
+	 * Return true if login is blocked globally (for all namespaces per instance - represented by namespace = null), false if not
+	 *
+	 * @param sess
+	 * @param login login to check
+	 */
+	boolean isLoginBlockedGlobally(PerunSession sess, String login);
+
+	/**
+	 * Return true if login is blocked for given namespace, false if not
+	 *
```

Comment:
Even though it is clear when we look on other methods, I would specify that globally blocked logins are not taken into account in this method (just to be sure).